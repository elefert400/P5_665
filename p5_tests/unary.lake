int x;
bool b;
int main(){
    int y;
    bool c;
    x = 1;
    b = true;
    y = -x;
    c = !b;
}