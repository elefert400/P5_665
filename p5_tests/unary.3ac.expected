[BEGIN GLOBALS]
x
b
[END GLOBALS]
[BEGIN main LOCALS]
y (local)
c (local)
tmp0 (tmp)
tmp1 (tmp)
[END main LOCALS]
enter main
x := 1
b := 1
tmp0 := NEG x
y := tmp0
tmp1 := NOT b
c := tmp1
lbl_0: EXIT